common/src/main/java/tachyon/worker/DataServerMessage.java

```diff
@@ -26,18 +26,23 @@
 
 import tachyon.Constants;
 import tachyon.network.protocol.RPCMessage;
+import tachyon.network.protocol.RPCResponse;
 
 /**
  * The message type used to send data request and response for remote data.
  */
 public class DataServerMessage {
-  public static final short DATA_SERVER_REQUEST_MESSAGE = 1;
-  public static final short DATA_SERVER_RESPONSE_MESSAGE = 2;
-
   private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
 
-  // TODO: convert client to netty to better deal with headers.
-  private static final int HEADER_LENGTH = 38;
+  // The size of the prefix of the header: frame length (long), messageType (int)
+  private static final int HEADER_PREFIX_LENGTH = 12;
+  // The request header is: HEADER_PREFIX, blockId (long), offset (long), length (long)
+  private static final int REQUEST_HEADER_LENGTH = HEADER_PREFIX_LENGTH + 24;
+  // The response header is: HEADER_PREFIX, blockId (long), offset (long), length (long),
+  // status (short)
+  private static final int RESPONSE_HEADER_LENGTH = HEADER_PREFIX_LENGTH + 26;
+  // The generic response header is: HEADER_PREFIX, status (short)
+  private static final int GENERIC_RESPONSE_HEADER_LENGTH = HEADER_PREFIX_LENGTH + 2;
```

Comment:
Maybe rename this as well? Though is this message only returned to indicate an error status? In that case maybe its better to just call it `ERROR_RESPONSE`